src/daft-parquet/src/read.rs

```diff
@@ -1,228 +1,122 @@
-
-                ptypes.push(&col.descriptor().descriptor.primitive_type);
-            }
-
-            // let field = &arrow_schema.fields[ii];
-            let arr_iter = column_iter_to_arrays(
-                decompressed_iters,
-                ptypes,
-                field.clone(),
-                Some(4096),
-                rg.num_rows(),
-            )?;
-
-            let all_arrays = arr_iter.collect::<arrow2::error::Result<Vec<_>>>()?;
-            let ser = all_arrays
-                .into_iter()
-                .map(|a| Series::try_from((field.name.as_str(), cast_array_for_daft_if_needed(a))))
-                .collect::<DaftResult<Vec<Series>>>()?;
-
-            daft_series[ii].extend(ser);
-        }
-    }
-
-    let compacted_series = daft_series
-        .into_iter()
-        .map(|s| Series::concat(s.iter().collect::<Vec<_>>().as_ref()))
-        .collect::<DaftResult<_>>()?;
-
-    Table::new(daft_schema, compacted_series)
-}
+use crate::{file::ParquetReaderBuilder, JoinSnafu};
 
 pub fn read_parquet(
     uri: &str,
     columns: Option<&[&str]>,
-    row_groups: Option<&[i64]>,
-    size: Option<usize>,
+    start_offset: Option<usize>,
+    num_rows: Option<usize>,
     io_client: Arc<IOClient>,
 ) -> DaftResult<Table> {
     let runtime_handle = get_runtime(true)?;
     let _rt_guard = runtime_handle.enter();
+    let (reader, ranges) = runtime_handle.block_on(async {
+        let builder = ParquetReaderBuilder::from_uri(uri, io_client.clone()).await?;
 
-    let (metadata, dl_ranges) = runtime_handle.block_on(async {
-        let size = match size {
-            Some(size) => size,
-            None => io_client.single_url_get_size(uri.into()).await?,
+        let builder = if let Some(columns) = columns {
+            builder.prune_columns(columns)?
+        } else {
+            builder
         };
+        let builder = builder.limit(start_offset, num_rows)?;
+        let parquet_reader = builder.build()?;
+        let ranges = parquet_reader.prebuffer_ranges(io_client.clone()).await?;
+        DaftResult::Ok((parquet_reader, ranges))
+    })?;
 
-        let metadata = read_parquet_metadata(uri, size, io_client.clone()).await?;
+    reader.read_from_ranges(ranges)
+}
 
-        let mut plan = plan_read_row_groups(uri, columns, row_groups, &metadata)?;
+pub fn read_parquet_schema(uri: &str, io_client: Arc<IOClient>) -> DaftResult<Schema> {
+    let runtime_handle = get_runtime(true)?;
+    let _rt_guard = runtime_handle.enter();
+    let builder = runtime_handle
+        .block_on(async { ParquetReaderBuilder::from_uri(uri, io_client.clone()).await })?;
+    Schema::try_from(builder.arrow_schema())
+}
 
-        plan.add_pass(Box::new(SplitLargeRequestPass {
-            max_request_size: 16 * 1024 * 1024,
-            split_threshold: 24 * 1024 * 1024,
-        }));
+pub fn read_parquet_statistics(uris: &Series, io_client: Arc<IOClient>) -> DaftResult<Table> {
```

Comment:
It's not a requirement for us to get the result of this operation as a `Table` (a `Vec<ParquetFileStatstics>` or `Vec<(int, int, str)>` could suffice) but if this avoids more Python glue code then should be fine